doc/tutorial/write_main_index.py

```diff
@@ -0,0 +1,38 @@
+import os
+import tempfile
+import glob
+import shutil
+# make the index page
+# I originally wanted to use jinja2 to do this, but the CSS has double curly
+# brackets, which jinja does not like, so here we go!
+examples_dir = os.path.join(tempfile.gettempdir(), 'scikit-beam-examples')
+output_dir = os.path.join(tempfile.gettempdir(), 'notebooks')
+if os.path.exists(output_dir):
+    shutil.rmtree(output_dir)
+    os.mkdir(output_dir)
+# copy all notebooks from the git repo to /tmp/notebooks
+nbs = glob.glob('/tmp/scikit-beam-examples/**/*.ipynb', recursive=True)
+
+for nb in nbs:
+    shutil.copy(nb, os.path.join(output_dir, os.path.split(nb)[1]))
+notebooks = os.listdir(output_dir)
+# read in the template
+files_to_convert = {'../index.rst.tmpl': '../index.rst',
+                    '../tutorial.rst.tmpl': '../tutorial.rst'}
+for template_file, output_file in files_to_convert.items():
+    with open(template_file, 'r') as f:
+        new_file = []
+        for line in f.readlines():
+            if line.endswith('\n'):
+                line = line[:-1]
+            # find the line where I want to write the list of examples
+            if 'REPLACE_WITH_LIST_OF_EXAMPLES' in line:
+                line = ['   tutorial/%s' % os.path.splitext(notebook_path)[0]
+                        for notebook_path in notebooks]
+                new_file.extend(line)
+            else:
+                new_file.append(line)
+
```

Comment:
No, it is correct as is currently written.  On line 32, the variable `line` is a list.  By calling `new_file.extend(line)`, I am adding all the elements in the list `line` to the other list, keeping `new_file` as a flat list.  If I just do `new_file.append`, then I will end up with a list of lists.  That being said, I see how this code looks confusing.  I will update the variable names